config/config.go

```diff
@@ -344,6 +344,28 @@ func (c *Configuration) EnsureFilePermissions() error {
 
 	r := regexp.MustCompile("^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}$")
 
+	// add trailing slash on data directory is no trailing slash exists
+	if ! strings.HasSuffix(c.System.Data, "/") {
+		c.System.Data = c.System.Data + "/"
+	}
+
+	// create the daemon-data dir if it doesn't exist
+	p, file := path.Split(c.System.Data)
+
+	if _, err := os.Stat(c.System.Data); err != nil {
+		// if file doesn't exist
+		if os.IsNotExist(err) {
+			//
+			if _, err = os.Stat(c.System.Data); err != nil {
```

Comment:
What is going on here? You have a nested stat call that is going to fail anyways since the first one failed?